tests/TalRepeatTest.php

```diff
@@ -143,8 +143,8 @@ function testTraversableRepeat()
         $doc->loadXML('<a><b/><c/><d/><e/><f/><g/></a>');
 
         $tpl = $this->newPHPTAL();
-        $tpl->setSource('<tal:block tal:repeat="node nodes"><tal:block tal:condition="php:repeat.node.index==4">(len=${repeat/node/length})</tal:block>${repeat/node/key}${node/tagName}</tal:block>');
-        $tpl->nodes = $doc->getElementsByTagName('*');
+        $tpl->setSource('<tal:block tal:repeat="node nodes"><tal:block tal:condition="php:repeat.node.index==4">(len=${nodes/length})</tal:block>${repeat/node/key}${node/tagName}</tal:block>');
```

Comment:
That's a bug in the previous implementation, fixed in PHP >=8.0, `length` is only in the root node